### PR TITLE
fix exception of curve.closest_point if no orthogonal projection

### PR DESCRIPTION
**Pull request by lottilotte**
if no orthogonal projection possible, use start or end point, whichever is closer

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_occ/geometry/curves/nurbs.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from compas.geometry import Frame
 from compas.geometry import Circle
 from compas.geometry import Box
+from compas.geometry._core.distance import distance_point_point
```
Comment (tomvanmele) [inline src/compas_occ/geometry/curves/nurbs.py]: `from compas.geometry import distance_point_point` (i assume this was an autoimport)

File: src/compas_occ/geometry/curves/nurbs.py [hunk 2]
```diff
@@ -601,10 +602,27 @@ def closest_point(self, point: Point, return_parameter: bool = False) -> Union[P
             The nearest as (point, parameter) tuple, if ``return_parameter`` is true.
         """
         projector = GeomAPI_ProjectPointOnCurve(point.to_occ(), self.occ_curve)
-        point = Point.from_occ(projector.NearestPoint())
+        try:
+            point = Point.from_occ(projector.NearestPoint())
+            if return_parameter:
+                parameter = projector.LowerDistanceParameter()
+
+        except RuntimeError:
```
Comment (tomvanmele) [inline src/compas_occ/geometry/curves/nurbs.py]: that is a very general error. can we be more specific?

**Review by tomvanmele — COMMENTED** [review]
small stuff.
otherwise looks good.
perhaps add a note about the modified behaviour?

File: src/compas_occ/geometry/curves/nurbs.py (hunk 1 above)
Comment (lottilotte) [reply src/compas_occ/geometry/curves/nurbs.py]: yes....

File: src/compas_occ/geometry/curves/nurbs.py (hunk 2 above)
Comment (lottilotte) [reply src/compas_occ/geometry/curves/nurbs.py]: > 

where should this note about the modified behaviour go? Inside the docstring?

Comment (tomvanmele) [reply src/compas_occ/geometry/curves/nurbs.py]: yes

File: src/compas_occ/geometry/curves/nurbs.py [hunk 3]
```diff
@@ -601,10 +602,29 @@ def closest_point(self, point: Point, return_parameter: bool = False) -> Union[P
...
         """
         projector = GeomAPI_ProjectPointOnCurve(point.to_occ(), self.occ_curve)
-        point = Point.from_occ(projector.NearestPoint())
+        try:
+            point = Point.from_occ(projector.NearestPoint())
+            if return_parameter:
+                parameter = projector.LowerDistanceParameter()
+
+        except RuntimeError as e:
+            if e.args[0] == 'StdFail_NotDoneGeomAPI_ProjectPointOnCurve::NearestPoint raised from method NearestPoint of class GeomAPI_ProjectPointOnCurve':
```
Comment (tomvanmele) [inline src/compas_occ/geometry/curves/nurbs.py]: or how about
```python
if args[0].startswith('StdFail_NotDoneGeomAPI'):
    ...
else:
    raise

**Review by tomvanmele — COMMENTED** [review]
almost there

File: src/compas_occ/geometry/curves/nurbs.py (hunk 2 above)
Comment (lottilotte) [reply src/compas_occ/geometry/curves/nurbs.py]: ok

**Review by tomvanmele — APPROVED** [review]

**Review by tomvanmele — APPROVED** [review]

**Comment by tomvanmele** [comment]
there is still some lint in your code
check the actions

**Pull request merged**